### ISSUE TITLE
Abstract platform-specific Stacktrace Functionality

### ISSUE THREAD

**Issue by fsimonis** [precice/precice#201]
The following headers in are linux-specific:
https://github.com/precice/precice/blob/3a27a4117937d2b979fceacf045d8edab2b580a5/src/utils/stacktrace.hpp#L3-L4

We should investigate whether [boost::stacktrace](https://www.boost.org/doc/libs/1_65_0/doc/html/stacktrace.html) could be used to implement this for all major platforms.

This is related to #200

**Comment by fsimonis** [comment]
Using boost::stacktrace the function `getStacktrace()` boils down to neat 3 lines:
https://github.com/precice/precice/blob/3a27a4117937d2b979fceacf045d8edab2b580a5/src/utils/stacktrace.hpp#L12-L62

```cpp
#include <boost/stacktrace.hpp>
static inline std::string getStacktrace()
{
    std::ostringstream strm;
    strm << boost::stacktrace::stacktrace();
    return strm.str(); 
}
```

**Comment by floli** [comment]
`boost::stacktrace` was addded in Boost 1.65. We have agreed on a 1.60 baseline for boost.

**Comment by fsimonis** [comment]
So for Windows we would have to do [this](https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/examining-the-stack-trace):

> To get a stack trace, use the methods GetStackTrace and GetContextStackTrace. A stack trace can be printed using OutputStackTrace and OutputContextStackTrace.

`execinfo.h` seems to be available in Mac OS X 10.5 and higher. So no problem there.

**Comment by floli** [comment]
In the changelog, the change is listed under 1.3.0, which is not correct, it should be under develop, which becomes 1.4.0 then.

**Comment by fsimonis** [comment]
@floli true about stacktrace, so I moved it.
The `MeshRequirement` for actions was merged before the release 1.3.0.
